cmake/deps.txt

```diff
@@ -23,10 +23,10 @@ microsoft_gsl;https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.zip;cf36
 microsoft_wil;https://github.com/microsoft/wil/archive/5f4caba4e7a9017816e47becdd918fcc872039ba.zip;fd119887d0d17c37adf1fc227b054befa28158ad
 mimalloc;https://github.com/microsoft/mimalloc/archive/refs/tags/v2.0.3.zip;e4f37b93b2da78a5816c2495603a4188d316214b
 mp11;https://github.com/boostorg/mp11/archive/refs/tags/boost-1.79.0.zip;c8f04e378535ededbe5af52c8f969d2dedbe73d5
-onnx;https://github.com/onnx/onnx/archive/1ba785612a79fe749aa1e478336e534743372639.zip;6c22c32fabafac81107186ea6bedb31741a3142d
+onnx;https://github.com/onnx/onnx/archive/refs/tags/v1.13.0.zip;8dda5079cdb5a134b08b0c73f4592a6404fc2dc6
```

Comment:
Do you need to change ONNX version? That will have big impacts.

Reply:
Because I came across a lot of exception like the below in Install ONNX step of windows CPU workflows.
[`libprotobuf.lib(coded_stream.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in cpp2py_export.obj [D:\a\_work\_temp\onnx\onnx-1ba785612a79fe749aa1e478336e534743372639\.setuptools-cmake-build\onnx_cpp2py_export.vcxproj`](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=860949&view=logs&j=ed9f9512-dfe4-5f72-a421-4225d265fdd2&t=356365da-9740-58aa-86bb-966853a34d70)
And it could be solved by upgrading ONNX with protobuf=3.20.2.

Reply:
And in Linux ci, there's the exception
#9 175.9 The conflict is caused by:
#9 175.9     The user requested protobuf==3.20.2
#9 175.9     onnx 1.12.0 depends on protobuf<=3.20.1 and >=3.12.2

Reply:
Why @liqunfu didn't use this commit in PR #13765 ?

Reply:
The link error is because at line https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/windows/helpers.ps1#L385 

we build protobuf with "-Dprotobuf_MSVC_STATIC_RUNTIME=OFF"

But in your PR it is not working. 

They error message you shared says that libprotobuf.lib was built with MT_StaticRelease but ONNX was built with MD_DynamicRelease. They should be the same.  It might because in a recent protobuf release they have changed their cmake options a bit.

Reply:
Anyway, ONNX 1.12 depends on protobuf 3.18.3, it looks that we have to upgrade it to meet compliance.